doc/examples/example1.md

```diff
@@ -1,139 +1,120 @@
-_eccKeyPairSect571r1 = AsymmetricKeypair::generate(mococrw::ECCSpec{openssl::ellipticCurveNid::SECT_571r1});
+mococrw::AsymmetricKeypair rsaKeyPair1024 = AsymmetricKeypair::generate(mococrw::RSASpec{1024});
 \endcode
 
-Key Writing to PEM
-==================
-
-Public key Writing to Pem
--------------------------
-
-Another functionality of the library is writing the key to a Pem doc.
-In this example it's being written a default RSA key pair to Pem.  
+Custom ECC Key
+--------------
 
+ECCSpec can be created by setting a custom curve type.
+List of all supported keys:
 \code{.cpp}
-const auto pemOfKey = _rsaKeyPair.publicKeyToPem();
+PRIME_192v1 = NID_X9_62_prime192v1,
+PRIME_256v1 = NID_X9_62_prime256v1,
+SECP_224r1 = NID_secp224r1,
+SECP_384r1 = NID_secp384r1,
+SECP_521r1 = NID_secp521r1,
+SECT_283k1 = NID_sect283k1,
+SECT_283r1 = NID_sect283r1,
+SECT_409k1 = NID_sect409k1,
+SECT_409r1 = NID_sect409r1,
+SECT_571k1 = NID_sect571k1,
+SECT_571r1 = NID_sect571r1,
 \endcode
-
-In this example it's being written a default ECC key pair to Pem. 
  
 \code{.cpp}
-const auto pemOfEccKey = _eccKeyPairDefault.publicKeyToPem();
+mococrw::AsymmetricKeypair eccKeyPairSect571r1 = AsymmetricKeypair::generate(mococrw::ECCSpec{openssl::ellipticCurveNid::SECT_571r1});
 \endcode
 
-In this example it's being written a ECC key pair with specifications to Pem.  
+Saving Asymmetric Keys in PEM format
+====================================
+
+Writing Public key to Pem
+-------------------------
+
+Another functionality of the library is writing the key to a Pem string.
+A default RSA key pair is being written to Pem. The process is the same
+regardless the key type.    
 
 \code{.cpp}
-const auto pemOfSectEccKey = _eccKeyPairSect571r1.publicKeyToPem();
+const std::string pemOfKey = rsaKeyPair.publicKeyToPem();
 \endcode
 
-Private key Writing to Pem
+Writing Private key to Pem
 --------------------------
 When writing a private key to a Pem we must give it a password. 
 \code{.cpp}
-const auto pemOfPrivateKey = eccPrivKey.privateKeyToPem("password");
+const std::string pemOfPrivateKey = eccPrivKey.privateKeyToPem("password");
 \endcode
 
 Key Reading from PEM
 ====================
 
-The user must generate the string manually. Or provide it from a document.  
-In this example, the following variables _pemEccPrivKeySect409k1 and _pemEccPubKeySect409k1 are strings which are going to hold the Private and the Public PEM's Key.
+Reading a Public Key from a PEM string
+--------------------------------------
 
-Private Key:
-\code{.cpp}
-std::string KeyHandlingTests::_pemEccPrivKeySect409k1{R"(-----BEGIN PRIVATE KEY-----
-MIHCAgEAMBAGByqGSM49AgEGBSuBBAAkBIGqMIGnAgEBBDQAF2zFhKyxJiI7bGvG
-Mw9rq7DUvrqTDJMHeRttpsZc0i9tFbvmaT2J5U39/RkseDha2b87oWwDagAEAAdj
-oVwkpy9CPA8RU3sd0aXV/XnHw5nE7HgINd6ApxCaknRebk4Vgbgz04588YqjqQpQ
-TAA+hxkUt1ZInurAHTt/ECQpvt1YOTBgNigakbLzq1LsbbyLWJsH5diall6Is+lg
-y2Mu1EA=
------END PRIVATE KEY-----)"};
-\endcode
+The user must generate the string manually. 
```

Comment:
This is not necessarily true and might be generated by the method above for some reason. So, probably we should remove this line to avoid confusion.